### PR TITLE
Add SignalR resource

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

File: sdk/provisioning/Azure.Provisioning/api/Azure.Provisioning.net6.0.cs [hunk 1]
```diff
@@ -410,6 +411,14 @@ public partial class ServiceBusTopic : Azure.Provisioning.Resource<Azure.Resourc
         public static Azure.Provisioning.ServiceBus.ServiceBusTopic FromExisting(Azure.Provisioning.IConstruct scope, string name, Azure.Provisioning.ServiceBus.ServiceBusNamespace parent) { throw null; }
     }
 }
+namespace Azure.Provisioning.SignalR
+{
+    public partial class SignalR : Azure.Provisioning.Resource<Azure.ResourceManager.SignalR.SignalRData>
```
Comment (tg-msft) [inline sdk/provisioning/Azure.Provisioning/api/Azure.Provisioning.net6.0.cs]: I think we need to give this a slightly longer name.  SignalRService?

File: sdk/provisioning/Azure.Provisioning/src/signalr/SignalRService.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        /// <param name="serviceMode">The service mode.</param>
+        /// <param name="parent">The parent.</param>
+        /// <param name="name">The name.</param>
+        /// <param name="version">The version.</param>
+        /// <param name="location">The location.</param>
+        public SignalRService(IConstruct scope, SignalRResourceSku? sku = default, IEnumerable<string>? allowedOrigins = default, string serviceMode = "Default", ResourceGroup? parent = default, string name = "signalr", string version = DefaultVersion, AzureLocation? location = default)
+            : this(scope, sku, parent, name, false, (name) => ArmSignalRModelFactory.SignalRData(
+                name: name,
+                location: location ?? Environment.GetEnvironmentVariable("AZURE_LOCATION") ?? AzureLocation.WestUS,
+                sku: sku ?? new SignalRResourceSku("Free_F1") { Capacity = 1 },
```
Comment (tg-msft) [inline sdk/provisioning/Azure.Provisioning/src/signalr/SignalRService.cs:37]: Is there a really constrained quota on free instances?  We might want to use the lowest paid option if so as the default in the CDK.

**Review by tg-msft — APPROVED** [review]

File: sdk/provisioning/Azure.Provisioning/src/signalr/SignalRService.cs (hunk 2 above)
Comment (JoshLove-msft) [reply sdk/provisioning/Azure.Provisioning/src/signalr/SignalRService.cs:37]: Yeah you are limited to 1 unit

Comment (JoshLove-msft) [reply sdk/provisioning/Azure.Provisioning/src/signalr/SignalRService.cs:38]: Discussed offline - the defaults will need to be reviewed across the board anyway and we want to make sure that Aspire is not relying on the defaults in general.

Comment (JoshLove-msft) [reply sdk/provisioning/Azure.Provisioning/src/signalr/SignalRService.cs:38]: /cc @mitchdenny

**Pull request merged**